packages/website/src/features/Router/RouterPage.tsx

```diff
@@ -0,0 +1,220 @@
+          <Link href="https://github.com/foundry-rs/foundry" isExternal>
+            Foundry
+          </Link>{' '}
+          project:
+        </Text>
+        <Box mb={4}>
+          <CommandPreview command="forge init" />
+        </Box>
+        <Text mb={4}>
+          This project will include a <Code>Counter.sol</Code> contract by
+          default. Duplicate this contract, rename it, and alter the function
+          names in it. For this example, we’ll assume you’ve rename the file and
+          contract to <Code>AnotherCounter</Code>.
+        </Text>
+        <Text mb={4}>
+          Create <Code>cannonfile.toml</Code> that deploys the two contracts and
+          a router:
+        </Text>
+        <Box mb={4}>
+          <CodePreview code={code1} language="toml" />
+        </Box>
+        <Text mb={4}>Build the Cannonfile:</Text>
+        <Box mb={4}>
+          <CommandPreview command="cannon build" />
+        </Box>
+        <Text mb={4}>
+          Run it. (By default, Cannon runs packages from the{' '}
+          <Link as={NextLink} href="/search">
+            package manager
+          </Link>
+          . Here, we add the <Code>--registry-priority local</Code> flag to
+          ensure we're using the version of this package that you just built,
+          regardless of what others have published.)
+        </Text>
+        <Box mb={4}>
+          <CommandPreview command="cannon sample-router-project --registry-priority local" />
+        </Box>
+        <Text mb={4}>
+          Press <Code>i</Code> to interact with the contracts in this project.
+          You'll see that the router contract exposes the functions from both
+          contracts.{' '}
+          <strong>
+            Always remember to interact with the router contract, and not the
+            dependent contracts directly.
+          </strong>
+        </Text>
+        <Heading size="md" mb={4} mt={6}>
+          Add Upgradability
+        </Heading>
+        <Text mb={4}>
+          We can also deploy a{' '}
+          <Link as={NextLink} href="/packages/transparent-upgradable-proxy">
+            transparent upgradeable proxy
+          </Link>{' '}
+          pointing at the router, such that this protocol can be upgradeable. In
+          the Cannonfile, add a setting for the admin (which will be allowed to
+          upgrade the proxy) and then provision the package which includes{' '}
+          <Link
+            href="https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/transparent/TransparentUpgradeableProxy.sol"
+            isExternal
+          >
+            the proxy contract
+          </Link>
+          :
+        </Text>
+        <Box mb={4}>
+          <CodePreview code={code2} language="toml" />
+        </Box>
+        <Text mb={4}>
+          If you alter one of your contracts, when building, Cannon will
+          automatically detect this, generate a new router, and upgrade the
```

Comment:
might also want to point out that any untouched contracts on the router will be recycled, saving precious gas. so you dont have to worry about the details of your deployment/what has changed. just focus on writing your changes, running deployments as necessary, and watch out for collisions.